mindmeld/models/nn_utils/input_encoders.py

```diff
@@ -0,0 +1,937 @@
+                    raise AssertionError(msg)
+
+        return examples, labels
+
+    @staticmethod
+    def _trim_combined(x: List[List[Any]], trim_len: int, y: List[Any] = None):
+        curr_len = 0
+        if y:
+            new_x, new_y = [], []
+            for _x, _y in zip(x, y):
+                if curr_len >= trim_len:
+                    return new_x, new_y
+                if curr_len + len(_x) > trim_len:
+                    new_x.append(_x[trim_len - curr_len])
+                    new_y.append(_y)
+                elif curr_len + len(_x) <= trim_len:
+                    new_x.append(_x)
+                    new_y.append(_y)
+            return new_x, new_y
+        else:
+            new_x = []
+            for _x in x:
+                if curr_len >= trim_len:
+                    return new_x
+                if curr_len + len(_x) > trim_len:
+                    new_x.append(_x[trim_len - curr_len])
+                elif curr_len + len(_x) <= trim_len:
+                    new_x.append(_x)
+            return new_x
+
+    def batch_encode(
+        self,
+        examples: Union[str, List[str]],
+        labels: Union[int, List[int]] = None,
+        padding_length: int = None,
+    ):
+        examples, labels = self._reformat_and_validate(examples, labels)
+
+        return_dict = {}
+        padding_length = padding_length or self.padding_length
+
+        if self.embedder_type == "bert":
+            # tokenize each word of each input seperately
+            tokenized_examples = [
+                [self._bert_tokenizer.tokenize(word) for word in example.split(" ")]
+                for example in examples
+            ]
+            # get maximum length of each example
+            max_curr_len = max([len(sum(t_ex, [])) for t_ex in tokenized_examples]) + 2  # cls, sep
+            # If padding_length is None, padding has to be done to the max length of the input batch
+            padding_length = min(max_curr_len, padding_length) if padding_length else max_curr_len
+
+            _trim_length = padding_length - 2  # -2 to sum upto padding_length after adding cls, sep
+            if labels:
+                trimmed_tokenized_examples, trimmed_labels = zip(*[
+                    self._trim_combined(t_ex, _trim_length, _labels)
+                    for t_ex, _labels in zip(tokenized_examples, labels)
+                ])
+            else:
+                trimmed_tokenized_examples = [
+                    self._trim_combined(t_ex, _trim_length) for t_ex in tokenized_examples
+                ]
+            split_lengths = [[len(x) for x in ex] for ex in trimmed_tokenized_examples]
+            trimmed_tokenized_examples = [" ".join(sum(ex, [])) for ex in
+                                          trimmed_tokenized_examples]
+            _inputs = self._bert_tokenizer(
+                trimmed_tokenized_examples, padding=True, truncation=True, max_length=None,
+                return_tensors="pt"
```

Comment:
what does `pt` mean?

Reply:
Common notation used in huggingface transformers.